packages/wrangler/src/config/validation-helpers.ts

```diff
@@ -14,16 +14,13 @@ export function deprecated<T extends object>(
   fieldPath: DeepKeyOf<T>,
   message: string,
   remove: boolean,
-  breaking = false
+  title = "🦺 DEPRECATION",
+  type: "warning" | "error" = "warning"
 ): void {
-  const diagonsticMessage = breaking
-    ? `🚨 NO LONGER SUPPORTED: "${fieldPath}":\n${message}`
-    : `🦺 DEPRECATION: "${fieldPath}":\n${message}`;
+  const diagnosticMessage = `${title}: "${fieldPath}":\n${message}`;
   const result = unwindPropertyPath(config, fieldPath);
   if (result !== undefined && result.field in result.container) {
-    (breaking ? diagnostics.errors : diagnostics.warnings).push(
-      diagonsticMessage
-    );
+    diagnostics[`${type}s`].push(diagnosticMessage);
```

Comment:
I just want to note how clever TS is now!! It is able to infer that the property names will be either `warnings` or `errors`.

Reply:
Love the use of the new release! Nice refactor 🎸